servicex_app/tests/test_code_gen_adapter.py

```diff
@@ -39,15 +38,31 @@ def _generate_test_request(self):
         return transform_request
 
     def test_init(self, mocker):
-        mock_transformer_manager = mocker.Mock()
+        mock_transformer_manager = mocker.MagicMock()
         service = CodeGenAdapter("http://foo.com", mock_transformer_manager)
         assert service.code_gen_url == "http://foo.com"
 
-    def test_generate_code_for_selection(self, mocker):
-        mock_response = mocker.Mock()
+    def test_generate_code_for_selection_with_params(self, mocker):
+        mock_response = mocker.MagicMock()
+        mock_response.status_code = 200
+        mock_requests_post = mocker.patch('requests.post', return_value=mock_response)
+        mock_transformer_manager = mocker.MagicMock()
+        mock_zip = mocker.patch("zipfile.ZipFile")
+        mocker.patch("io.BytesIO")
+        service = CodeGenAdapter("http://foo.com", mock_transformer_manager)
+        mock_transformer_image = "testimage"
+        service.generate_code_for_selection(self._generate_test_request(), "servicex", mock_transformer_image)
+        mock_requests_post.assert_called()
+
+        mock_transformer_manager.create_configmap_from_zip.assert_called_with(mock_zip(),
+                                                                              "462-33",
+                                                                              "servicex")
+
+    def test_generate_code_for_selection_without_params(self, mocker):
```

Comment:
We need tests for exercising the Multipart mime return (`transformer_image=None`)

Reply:
why would multipart mime return none? I am a bit confused on that, of the two cases we discussed.
1. App sends the transformer image to codegen, then codegen will return that image.
2. If app does not send the transformer image, the codegen returns the default config image.

In either case transformer image will not be None

Reply:
Ok - I see what this argument is now... is there a good reason to send it to the code generator? Couldn't we allow the code gen to send back something but ignore it and use this provided one entirely in the App?

Also, we don't have any tests of the multipart mime unpacking